### ISSUE TITLE
InfluxdbWriter chokes on perfdata "U"

### ISSUE THREAD

**Issue by joernott** [Icinga/icinga2#9282]
## Describe the bug

I am getting a lot of  warnings in my icinga2 logs like this:
```
warning/InfluxdbWriter: Ignoring invalid perfdata for checkable 'dev-bas1-pos-v01.sf-rz.de!postgres_patroni_dev_01_tst_longest_query' and command 'check_pgactivity' with value: 'testdb max'=Us;60;90
```
Looking at https://nagios-plugins.org/doc/guidelines.html#AEN200, the value "U" is perfectly legitimate here

## To Reproduce

1. set up a check which will return "U" as value in the performancedata (check_pgactivity works fine with the longest_query)
2. run the check and look into the logs

## Expected behavior
I would love to turn off those warnings in the icinga logs (only for value U) or have InfluxWriter insert something like null instead of failing/complaining.

## Your Environment

Include as many relevant details about the environment you experienced the problem in

* Version used (`icinga2 --version`): 2.13.2-1
* Operating System and version: RHEL 7.9
* Enabled features (`icinga2 feature list`): api checker ido-pgsql influxdb mainlog notification

**Comment by joernott** [comment]
Yes, that is what the check is sending and Icinga is passing to the writer. So I would expect it to handle that value without spamming the logs.

**Comment by easterhanu** [comment]
Happens with other plugins too, e.g. `check_ping` when attempting to ping an unaccessible host:
```
# ./check_ping -H 10.1.2.3 -w 3000,80% -c 5000,100%
PING CRITICAL - Packet loss = 100%| rta=U;3000.000000;5000.000000;; pl=100%;80;100;0;
```
"U" is a valid value (unknown unit of measurement) in perf data, so should not cause an `Ignoring invalid perfdata` error.

**Comment by slalomsk8er** [comment]
> I would love to turn off those warnings in the icinga logs (only for value U) or have InfluxWriter insert something like null instead of failing/complaining.

As long as the other values of the check are processed I would expect it to just drop the one and log it as info and not as warning.
Alternatively one could keep it and drop it later depending on if the perfdata writer in question supports a null value for the entry and log it there as info.

**Comment by oxzi** [comment]
FTR, this issue is related to #9038.

**Comment by julianbrost** [comment]
That particular log message is coming from this piece of code:

https://github.com/Icinga/icinga2/blob/333534096ebc251de08cecd2f688bc690253902c/lib/perfdata/influxdbcommonwriter.cpp#L276-L284

So what are the options here?

1. If the value is `U`, just discard the whole perfdata "value" as it has no value so that it never reaches the writers. Not sure if there are any other consumers that would then miss it if we discard it early.
2. Make a difference in error handling so that `U` is silently ignored, while actually invalid perfdata still triggers a log message.
3. Treat is as valid perfdata. I already see two things here that would probably make this the most complex approach:
    - That pushes the question of what to do with it down to the individual consumers. In particular, every writer and corresponding backend needs to be checked how this can be communicated. If we just send some `null`/`NaN` value, it's quite possible that it would break in other ways.
    - The example from [a previous comment](https://github.com/Icinga/icinga2/issues/9282#issuecomment-3306785625), `rta=U;3000.000000;5000.000000;;`, raises another question with actually treating it as valid perfdata: how should it be treated if the unit information is missing, in particular with Icinga 2 trying to normalize the values to a base unit? Or is this actually an issue with `check_ping` and it should better write `Ums` instead of just `U` so that it's still clear how the limits and thresholds are supposed to be interpreted?

Also note that this isn't specific to the InfluxDB writer, but the same snippets exists in other writers as well (that's not an in-depth check, just the result of a very basic `git grep 'Ignoring invalid perfdata for checkable'`):

https://github.com/Icinga/icinga2/blob/333534096ebc251de08cecd2f688bc690253902c/lib/perfdata/elasticsearchwriter.cpp#L199-L207
https://github.com/Icinga/icinga2/blob/333534096ebc251de08cecd2f688bc690253902c/lib/perfdata/gelfwriter.cpp#L323-L331
https://github.com/Icinga/icinga2/blob/333534096ebc251de08cecd2f688bc690253902c/lib/perfdata/graphitewriter.cpp#L344-L352
https://github.com/Icinga/icinga2/blob/333534096ebc251de08cecd2f688bc690253902c/lib/perfdata/opentsdbwriter.cpp#L327-L335
